### ISSUE TITLE
periodic BCs incompatible with netcdf topo files

### ISSUE THREAD

**Issue by bolliger32** [clawpack/geoclaw#539]
I'm having problems running a global model with periodic boundary conditions in the E-W direction and "extra" conditions in the N-S direction. I believe this has to do with how netcdf's are loaded [here](https://github.com/clawpack/geoclaw/blob/b005768de402bee90a8340b4ac18fab2428b3b8e/src/2d/shallow/topo_module.f90#L951), b/c they are clipped to within a `dx` and `dy` buffer of the domain. This means there is no topography loaded that lies outside of the domain within the ghost cell portion.

This comes into play in [setaux.f90](https://github.com/clawpack/geoclaw/blob/b005768de402bee90a8340b4ac18fab2428b3b8e/src/2d/shallow/setaux.f90#L151). When setting for ghost cells, `wrap_coords` is called, which, as expected, wraps the E-W cells but does nothing to the N-S cells (since `yperdom` is false).

Then `cellgridintegrate` is called, which due to [this line](https://github.com/clawpack/geoclaw/blob/b005768de402bee90a8340b4ac18fab2428b3b8e/src/2d/shallow/cellgridintegrate2.f#L77), raises an error that there is no topography overlapping that cell.

Does that sound like the right culprit (cc: @mandli @rjleveque)? I imagine the fix would be to increase the extent of the netcdf that gets loaded if you are in this situation.

**Comment by mandli** [comment]
Makes sense to me.  Periodic BCs are not used in GeoClaw except for planetary wide simulations and I am not sure we have used anything but the ESRI ASCII formats.
